modules/storages/lib/api/v3/utilities/storage_interaction/nextcloud_storage_query.rb

```diff
@@ -0,0 +1,134 @@
+module API::V3::Utilities::StorageInteraction
+  class NextcloudStorageQuery
+    def initialize(base_uri:, origin_user_id:, token:, with_refreshed_token:)
+      @uri = base_uri
+      @origin_user_id = origin_user_id
+      @token = token
+      @with_refreshed_token = with_refreshed_token
+      @base_path = "/remote.php/dav/files/#{@origin_user_id}/"
+    end
+
+    def files
+      http = Net::HTTP.new(@uri.host, @uri.port)
+      http.use_ssl = @uri.scheme == 'https'
+
+      result = @with_refreshed_token.call do
+        response = http.propfind(
+          @base_path,
+          requested_properties,
+          {
+            'Depth' => '1',
+            'Authorization' => "Bearer #{@token}"
+          }
+        )
+
+        response.is_a?(Net::HTTPSuccess) ? ServiceResult.success(result: response.body) : error(response)
+      end
+
+      storage_files(result)
+    end
+
+    private
+
+    def requested_properties
+      Nokogiri::XML::Builder.new do |xml|
+        xml['d'].propfind(
+          'xmlns:d' => 'DAV:',
+          'xmlns:oc' => 'http://owncloud.org/ns'
+        ) do
+          xml['d'].send('prop') do
+            xml['oc'].send('fileid')
+            xml['oc'].send('size')
+            xml['d'].send('getcontenttype')
+            xml['d'].send('getlastmodified')
+            xml['oc'].send('owner-display-name')
+          end
```

Comment:
I used send here to make explicit, that these are not actual methods defined on the XML object. The implementation uses `method_missing` to provide this API for the actual creation of the sub-elements by it's provided name (string). I can change it back, but it's easier to see this way IMO.

Reply:
IMO a caller should not care for the implementation details of the method being called. In the end, whether `method_missing` is used to get the response or not does not alter the contract between caller and callee. 

`send` is typically used for two purposes in ruby that I am aware of:
 * In metaprogramming, so when the method called is determined programatically. This is a legitimate use case.
 * When calling a private/protected method. This has a strong smell and should only be used if there is a compelling  reason to do so.

Since the strings above are static, to me, the sending looks like the second case even if it isn't in this case.

Reply:
While refactoring it, I came across `owner-display-name`. It has to be done with `send`. I think therefor I converted all to `send` to not confuse anyone in the PR ;).

Reply:
```
xml['d'].prop do
            xml['oc'].fileid
            xml['oc'].size
            xml['d'].getcontenttype
            xml['d'].getlastmodified
            xml['oc'].send('owner-display-name')
```